extensions/blocks/podcast-episodes/podcast-episodes.php

```diff
@@ -20,24 +22,166 @@
 function register_block() {
 	jetpack_register_block(
 		BLOCK_NAME,
-		array( 'render_callback' => __NAMESPACE__ . '\load_assets' )
+		array(
+			'attributes'      => array(
+				'url'         => array(
+					'type' => 'url',
+				),
+				'itemsToShow' => array(
+					'type'    => 'integer',
+					'default' => 5,
+				),
+			),
+			'render_callback' => __NAMESPACE__ . '\render_block',
+		)
 	);
 }
 add_action( 'init', __NAMESPACE__ . '\register_block' );
 
 /**
  * Podcast Episodes block registration/dependency declaration.
  *
- * @param array  $attr    Array containing the Podcast Episodes block attributes.
- * @param string $content String containing the Podcast Episodes block content.
+ * @param array  $attributes Array containing the Podcast Episodes block attributes.
+ * @param string $content    String containing the Podcast Episodes block content.
  *
  * @return string
  */
-function load_assets( $attr, $content ) {
-	/*
-	 * Enqueue necessary scripts and styles.
+function render_block( $attributes, $content ) {
+
+	if ( empty( $attributes['url'] ) ) {
+		return;
+	}
+
+	$track_list = get_track_list( $attributes['url'], $attributes['itemsToShow'] );
+
+	if ( is_wp_error( $track_list ) ) {
+		return '<p>Unable to retrieve track list. Please check your Podcast feed URL.</p>';
+	}
+
+	return render_player( $track_list );
+}
+
+/**
+ * Renders the HTML for the Podcast player and tracklist.
+ *
+ * @param array $track_list the list of podcast tracks.
+ * @return string the HTML for the podcast player.
+ */
+function render_player( $track_list ) {
+	global $content_width;
+
+	$player_data = array(
+		'type'         => 'audio',
+		// Don't pass strings to JSON, will be truthy in JS.
+		'tracklist'    => true,
+		'tracknumbers' => true,
+		'images'       => true,
+		'artists'      => true,
+		'tracks'       => $track_list,
+	);
+
+	$outer         = 22; // Default padding and border of wrapper.
+	$default_width = 640;
+	$theme_width   = empty( $content_width ) ? $default_width : ( $content_width - $outer );
+
+	// If there are no tracks (it is possible) then display appropriate user facing error message.
+	if ( empty( $track_list ) ) {
+		return '<p>No tracks available to play.</p>';
+	}
+
+	ob_start();
+	wp_playlist_scripts( 'audio' );
+	/**
+	 * Prints and enqueues playlist scripts, styles, and JavaScript templates.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @param string $type  Type of playlist. Possible values are 'audio' or 'video'.
+	 * @param string $style The 'theme' for the playlist. Core provides 'light' and 'dark'.
 	 */
+	do_action( 'wp_playlist_scripts', 'audio', 'light' );
+
+	?>
+	<div class="wp-block-<?php echo esc_attr( BLOG_SLUG ); ?> wp-playlist wp-audio-playlist wp-playlist-light">
+		<div class="wp-playlist-current-item"></div>
+		<audio src="<?php echo esc_url( $track_list[0]['src'] ); ?>" controls="controls" preload="none" width="<?php echo esc_attr( (int) $theme_width ); ?>"></audio>
```

Comment:
just FYI: I've been testing and some lists don't have index [0] 😱. I need to improve it but totally ok in the following one.

Reply:
It shoudn't be possible as I strip out all tracks without a `src` but it's worth guarding I guess.

Reply:
Fixed.

Reply:
I think there might have been a mix-up in bb8248a0a2c6eb14a6b26c88c6acd88cd80688ef. `$initial_track_src` was correctly created but it was used on a wrong line instead of this one. Do I read it correctly?